pkg/ignition/ignition.go

```diff
@@ -60,9 +61,33 @@ func New(c client.Client) (*Ignition, error) {
 	}
 	log.V(1).Info("parsed", "machineconfig", renderedWorker.GetName(), "ignition version",
 		configuration.Ignition.Version)
+
+	ccList := mcfg.ControllerConfigList{}
+	if err := c.List(context.TODO(), &ccList); err != nil {
+		return nil, err
+	}
+	var kubeletCAData []byte
+	for _, item := range ccList.Items {
+		if item.Spec.KubeAPIServerServingCAData != nil {
+			log.V(1).Info("processing kubelet-ca", "ControllerConfig", item.Name)
+			kubeletCAData = item.Spec.KubeAPIServerServingCAData
+			break
+		}
+	}
+	if len(kubeletCAData) == 0 {
+		return nil, fmt.Errorf("cannot find kubelet-ca")
+	}
```

Comment:
Not asking for a change but this sort of changes should be in a helper function.

Reply:
Agree. I'll take it for a follow-up PR.